### DISCUSSION TITLE
Runtime assertion failure while using ImageMagick

### DISCUSSION THREAD

**Discussion by cparkhi16** [ImageMagick/ImageMagick#6493] [Help]
### ImageMagick version

7.1.1-11

### Operating system, version and so on

Linux

### Description

I am currently using MagickWand to read a input image and convert it to webp format , I am successfully able to do it for most of the images but sometimes my program crashes and it gives following assertion failed errors: 

**MagickCore/magick.c:449: GetMagickDecoderThreadSupport: Assertion `magick_info->signature == MagickCoreSignature' failed.
MagickCore/semaphore.c:295: LockSemaphoreInfo: Assertion `semaphore_info != (SemaphoreInfo *) NULL' failed.**

**Comment by dlemstra** [comment]
Can you please explain why you closed this issue? This might help someone in the future.

**Reply by cparkhi16** [reply]
Hi
I thought that this issue is because the way I am using ImageMagick and hence marked it as resolved.
Actually while cleaning up the resources I was facing this issue and I thought that I was taking a wrong approach.
When I avoided calling "DestroyMagickWand" ,I was not getting those assertion errors.

`void cleanupImageMagick(MagickWand *inputImage, MagickWand *snapshot)
{
    if (snapshot)
    {
        snapshot = DestroyMagickWand(snapshot);
    }
    if (inputImage)
    {
        inputImage = DestroyMagickWand(inputImage);
    }
    MagickWandTerminus();
}`
